### ISSUE TITLE
Compiling with ICU : icu-config deprecated?

### ISSUE THREAD

**Issue by jmferrell** [pennmush/pennmush#1419]
I'm playing around with compiling on Ubuntu 22.04 with ICU support, but it looks like it's failing because the configure script is looking for icu-config, which has apparently been deprecated in favor of pkg-config now. Is there a possibly workaround to get configure to look for pkg-config? Or would I need to downgrade my ICU version?

**Comment by depwl9992** [comment]
Unable to test unicode in my MUSH, but this at least got rid of the warning and got `Enable Unicode Support ... yes` in the report when I ran `./configure`.

```
apt install pkg-config libpq-dev
ln -s /usr/bin/pkg-config /usr/bin/icu-config
./configure
```

**Comment by jmferrell** [comment]
Hey, thank! That symlink did the trick!

**Comment by jmferrell** [comment]
Ugh. Or maybe not. ./configure shows
`Enable Unicode support ............. : yes`
but  also
`checking for icu-config... /usr/bin/icu-config`
`checking for ICU >= 51... ICU not found`
So it's still not actually loading ICU somehow?

**Comment by depwl9992** [comment]
Ugh. I'm noticing this too, and running up against a wall.

Tried editing `./configure` to allow pkg-config versions >= 1.8.1 (line 10925), but upon running the script to reconfigure I get:
```
checking for icu-config... /usr/bin/icu-config                                   
checking for ICU >= 1.8.1... yes                                                 
checking ICU_CPPFLAGS... pkgconf: unknown option -- cppflags                     
                                                                                 
checking ICU_CFLAGS... Please specify at least one package name on the command li
ne.                                                                              
                                                                                 
checking ICU_CXXFLAGS... pkgconf: unknown option -- cxxflags                     
                                                                                 
checking ICU_LIBS... pkgconf: unknown option -- ldflags
```

[This "fun" SO discussion](https://stackoverflow.com/questions/46550299/why-doesnt-pkg-config-support-cxxflags) talks about these errors in pkgconf which they reference setting some compile flags for C99 & C++11, which reminds me of an old discussion I remember our devs having over a decade ago and saying something along the lines of, "to update the project to C++11 is really going to need an entire code rewrite almost from scratch. Not worth it/it's on the rearmost back burner we can stick it on."

I have a feeling this may be a brute-force method that...won't amount to much without really understanding how they're using icu-config in the first place. Unless there's an old, drop-in replacement for it available, or it can be installed from a legacy apt package without causing other issues..

And people online have the same sorts of complaints it seems for other projects still needing it..

**Comment by depwl9992** [comment]
For reference, [here's the discussion](https://stackoverflow.com/questions/62108757/icu-config-not-works-but-already-installed) that initially gave me the symbolic link method, answered by the [author](https://manpages.ubuntu.com/manpages/xenial/man1/icu-config.1.html) of icu-config himself.

**Comment by jmferrell** [comment]
Ah, cool—thanks for the references, at least. Just something to keep an eye on, I guess. I was looking for a way to display accented characters on my MUSH . . . am I even barking up the right tree, then?

**Comment by depwl9992** [comment]
OH!! Accented characters I think _may_ already be supported without Unicode (since a good number of them are part of the extended ASCII characterset). I've seen them used on various MUSHes over the years in varying degrees. But I've not been able to get them to work outright on mine. That said, getting Unicode to work would probably help the attempt pretty substantially, as on mine I'm not seeing the accent() function working at all, but also don't have NOACCENT set or anything like that.

It may be a client-side issue, but I'll see if I can do some digging and figure out what they used to say about getting it to work before without involving ICU.

**Comment by jmferrell** [comment]
Yeah, I feel like I was running in a circle at some point, if I remember correctly: It seemed like accents should be possible, since they were part of Latin-1, but maybe Unicode would fix that? But Unicode needed ICU, which wouldn't work, but maybe it was just a matter of changing the internationalization? But then I feel like I found that that relied on ICU somehow, as well... ended up down so many rabbit holes I finally gave up (mostly).

**Comment by HarryCordewener** [comment]
Accented characters can be enabled by setting the `LANG` in `restart`.
